### DISCUSSION TITLE
State management

### DISCUSSION THREAD

**Discussion by data-handler** [redom/redom#341] [General]
Does anyone use a state management library alongside re:dom? Does re:dom lend itself to using a separate state management system?

**Comment by pakastin** [comment]
RE:DOM doesn't do any black magic, so it's very easy to use with any state handling library.

**Comment by joona** [comment]
I'm a bit late to the party, but it's quite easy to use external state management with RE:DOM.

Here's a gist of really simplified state management toolset that I'm using myself. I'm sorry, it lacks usage examples and documentation. 
https://gist.github.com/joona/cc583c14cf1d3c2749867499ec912881

It's really simplified implementation of FLUX pattern, which can be used pretty flexibly in terms of how you want to organize your code. The main idea is that **Stores** to hold the single source of truth and propagate it to RE:DOM components. Listeners for store updates can be bound easily to RE:DOM components on `onmount`, `onunmount` hooks and update events can be mapped to `update()` or some other method to propagate state changes. UI or API interactions **dispatch** and execute **Actions**, which themselves are only constant strings in the codebase. **Store** state is manipulated only through **action handlers**, which can be simple switch statements or wrapped to functions like in redux, i.e.

It also uses DOM as event bus with `CustomEvent`s, so no external event handling libraries are needed.
